src/components/ha-qr-scanner.ts

```diff
@@ -88,29 +91,37 @@ class HaQrScanner extends LitElement {
     this._loadQrScanner();
   }
 
-  protected updated(changedProps: PropertyValues) {
-    if (changedProps.has("error") && this.error) {
-      alert(`error: ${this.error}`);
-      this._notifyExternalScanner(this.error);
-    }
-  }
-
   protected render() {
-    if (this._nativeBarcodeScanner && !this._manual) {
+    if (this._nativeBarcodeScanner) {
       return nothing;
     }
 
-    return html`${this.error
-      ? html`<ha-alert alert-type="error">${this.error}</ha-alert>`
-      : ""}
-    ${navigator.mediaDevices && !this._manual
+    return html`${this._error || this._warning
+      ? html`<ha-alert .alertType=${this._error ? "error" : "warning"}>
+          ${this._error || this._warning}
+          ${this._error
+            ? html` <ha-button @click=${this._retry} slot="action">
+                ${this.hass.localize("ui.components.qr-scanner.retry")}
+              </ha-button>`
+            : nothing}
+        </ha-alert>`
+      : nothing}
+    ${navigator.mediaDevices
       ? html`<video></video>
           <div id="canvas-container">
-            ${this._cameras && this._cameras.length > 1
+            ${this._loading
+              ? html`<div class="loading">
+                  <ha-spinner active></ha-spinner>
+                </div>`
+              : nothing}
+            ${!this._loading &&
+            !this._error &&
```

Comment:
I don't think we should hide the scanner and have the user click retry if we have not found a barcode in time?

Reply:
Maybe make that a warning instead of an error

Reply:
Or just keep going? I find showing an error a bit annoying, especially when some of the Z-Wave QR codes are tiny and barely scanable.

Reply:
@bramkragten we have `this._qrNotFoundCount` that stops the scanner when it's at `250`. For me it was strange to have a warning and the scanner still running but it stopped to scan.

Do you know why we have this limit of `250`? If it's possible we can remove the limit and so we don't have to do this retry thing.

Reply:
I don't think it stopped scanning? If it did that was a bug.

There is no specific reason for the 250, the library we use sends it way too often, so I throttled it...

Reply:
Oh ok, yes then I'll try to fix it and of course just show a warning.

Reply:
Again, do we even need any warning or so at all? Either a QR code is detected or not. IMO being reminded that no QR code is found seems unnecessary.